adapters/repos/db/index_integration_test.go

```diff
@@ -15,218 +15,238 @@
-// 	require.Nil(t, err)
-
-// 	afterRecreateObj2, err := index.objectByID(context.TODO(), productsIds[1], nil, additional.Properties{})
-// 	require.Nil(t, err)
-
-// 	// insert some data in the recreated index
-// 	for i, p := range products {
-// 		thing := models.Object{
-// 			Class:      testClass.Class,
-// 			ID:         productsIds[i],
-// 			Properties: p,
-// 		}
-
-// 		err := index.putObject(context.TODO(), storobj.FromObject(&thing, []float32{0.1, 0.2, 0.01, 0.2}))
-// 		require.Nil(t, err)
-// 	}
-
-// 	afterRecreateAndInsertObj1, err := index.objectByID(context.TODO(), productsIds[0], nil, additional.Properties{})
-// 	require.Nil(t, err)
-
-// 	afterRecreateAndInsertObj2, err := index.objectByID(context.TODO(), productsIds[1], nil, additional.Properties{})
-// 	require.Nil(t, err)
-
-// 	assert.Equal(t, 3, len(indexFilesBeforeDelete))
-// 	assert.Equal(t, 0, len(indexFilesAfterDelete))
-// 	assert.Equal(t, 3, len(indexFilesAfterRecreate))
-// 	assert.Equal(t, indexFilesBeforeDelete, indexFilesAfterRecreate)
-// 	assert.NotNil(t, beforeDeleteObj1)
-// 	assert.NotNil(t, beforeDeleteObj2)
-// 	assert.Empty(t, afterRecreateObj1)
-// 	assert.Empty(t, afterRecreateObj2)
-// 	assert.NotNil(t, afterRecreateAndInsertObj1)
-// 	assert.NotNil(t, afterRecreateAndInsertObj2)
-// }
+func TestIndex_DropIndex(t *testing.T) {
+	dirName := t.TempDir()
+	class := &models.Class{Class: "deletetest"}
+	index := emptyIdx(t, dirName, class)
+
+	indexFilesBeforeDelete, err := getIndexFilenames(dirName, class.Class)
+	require.Nil(t, err)
+
+	err = index.drop()
+	require.Nil(t, err)
+
+	indexFilesAfterDelete, err := getIndexFilenames(dirName, class.Class)
+	require.Nil(t, err)
+
+	assert.Equal(t, 5, len(indexFilesBeforeDelete))
+	assert.Equal(t, 0, len(indexFilesAfterDelete))
+}
+
+func TestIndex_DropIndexReadOnly(t *testing.T) {
+	ctx := testCtx()
+	class := &models.Class{Class: "deletetest"}
+	shard, index := testShard(t, ctx, class.Class)
+
+	err := index.updateShardStatus(ctx, shard.name, storagestate.StatusReadOnly.String())
+	require.Nil(t, err)
+
+	err = index.drop()
+	require.Nil(t, err)
+}
+
+func TestIndex_DropEmptyAndRecreateEmptyIndex(t *testing.T) {
+	dirName := t.TempDir()
+	class := &models.Class{Class: "deletetest"}
+	index := emptyIdx(t, dirName, class)
+
+	indexFilesBeforeDelete, err := getIndexFilenames(dirName, class.Class)
+	require.Nil(t, err)
+
+	// drop the index
+	err = index.drop()
+	require.Nil(t, err)
+
+	indexFilesAfterDelete, err := getIndexFilenames(dirName, class.Class)
+	require.Nil(t, err)
+
+	index = emptyIdx(t, dirName, class)
+
+	indexFilesAfterRecreate, err := getIndexFilenames(dirName, class.Class)
+	require.Nil(t, err)
+
+	assert.Equal(t, 5, len(indexFilesBeforeDelete))
+	assert.Equal(t, 0, len(indexFilesAfterDelete))
+	assert.Equal(t, 5, len(indexFilesAfterRecreate))
+
+	err = index.drop()
+	require.Nil(t, err)
+}
+
+func TestIndex_DropWithDataAndRecreateWithDataIndex(t *testing.T) {
```

Comment:
Not sure if possible, but could you set the sahrd to readonly before dropping it?

Reply:
done